mslice/cli/projection_functions.py

```diff
@@ -1,46 +1,108 @@
-    workspace = get_workspace_handle(InputWorkspace)
+    _check_workspace_name(workspace)
+    workspace = get_workspace_handle(workspace)
     if not isinstance(workspace, HistogramWorkspace):
         raise RuntimeError("Incorrect workspace type.")
-    if IntensityStart == 0 and IntensityEnd == 0:
-        intensity_range = None
-    else:
-        intensity_range = (IntensityStart, IntensityEnd)
-    lines = cli_cut_plotter_presenter.plot_cut_from_workspace(workspace, intensity_range=intensity_range,
-                                                              plot_over=PlotOver)
 
-    return lines
+    presenter = get_cut_plotter_presenter()
+
+    plot_over = kwargs.pop('plot_over', True)
+    intensity_range = kwargs.pop('intensity_range', (None, None))
+    x_units = kwargs.pop('x_units', 'None')
+    legend = kwargs.pop('label', None)
+    legend = workspace.name if legend is None else legend
+
+    plotfunctions.errorbar(axes, workspace.raw_ws, label=legend, *args, **kwargs)
 
+    axes.set_ylim(*intensity_range) if intensity_range is not None else axes.autoscale()
+    if cur_canvas.manager.window.action_toggle_legends.isChecked():
+        leg = axes.legend(fontsize='medium')
+        leg.draggable()
+    axes.set_xlabel(get_display_name(x_units, get_comment(workspace)), picker=CUT_PICKER_TOL_PTS)
+    axes.set_ylabel(CUT_INTENSITY_LABEL, picker=CUT_PICKER_TOL_PTS)
+    if not plot_over:
+        cur_canvas.set_window_title(workspace.name)
+        cur_canvas.manager.update_grid()
+    if not cur_canvas.manager.has_plot_handler():
+        plot_handler = cur_canvas.manager.add_cut_plot(presenter, workspace.name.rsplit('_', 1)[0])
+        plot_handler.save_default_options()
+    cur_fig.canvas.draw()
 
-def PlotSliceMsliceProjection(InputWorkspace, IntensityStart="", IntensityEnd="", Colormap=DEFAULT_CMAP):
+    return axes.lines
+
+
+@plt.set_category(plt.CATEGORY_SLICE)
+def pcolormesh(axes, workspace, *args, **kwargs):
     """
     Same as the CLI PlotSlice but returns the relevant axes object.
     """
-
-    _check_workspace_name(InputWorkspace)
-    workspace = get_workspace_handle(InputWorkspace)
+    from mslice.app.presenters import get_slice_plotter_presenter, cli_slice_plotter_presenter
+    _check_workspace_name(workspace)
+    workspace = get_workspace_handle(workspace)
     _check_workspace_type(workspace, HistogramWorkspace)
 
     # slice cache needed from main slice plotter presenter
     if is_gui():
         cli_slice_plotter_presenter._slice_cache = app.MAIN_WINDOW.slice_plotter_presenter._slice_cache
-    cli_slice_plotter_presenter.change_intensity(workspace.name, IntensityStart, IntensityEnd)
-    cli_slice_plotter_presenter.change_colourmap(workspace.name, Colormap)
-    quadmesh = cli_slice_plotter_presenter.plot_from_cache(workspace)
+    else:
+        # Needed so the figure manager knows about the slice plot handler
+        create_slice_figure(workspace.name[2:], get_slice_plotter_presenter())
```

Comment:
So the issue with the plot having no `plot_handler` for the generated CLI commands when executed in the GUI comes from this part here. The `plot_handler` reference is only created in this call to `create_slice_figure` which is not run if `is_gui` is `True`. I suggest changing line 72 to:

```
    if is_gui() and GlobalFigureManager.get_active_figure()._plot_handler is not None:
```

Reply:
Ah I see. I'm looking into it now

Reply:
Yes I was able to reproduce the error. And this fixes it. Thank you.